src/game/chraicommands.c

```diff
@@ -5125,22 +5355,45 @@ bool aiRevokeControl(void)
 	struct chrdata *chr = chrFindById(g_Vars.chrdata, cmd[2]);
 
 	if (chr && chr->prop && chr->prop->type == PROPTYPE_PLAYER) {
-		u32 prevplayernum = g_Vars.currentplayernum;
-		u32 playernum = playermgrGetPlayerNumByProp(chr->prop);
-		setCurrentPlayerNum(playernum);
-		bgunSetSightVisible(GUNSIGHTREASON_NOCONTROL, false);
-		bgunSetGunAmmoVisible(GUNAMMOREASON_NOCONTROL, false);
+		if (isChrPropCoop(chr->prop)) {
+			struct playerpool** playerpool = getPlayerPool(CHR_COOP);
```

Comment:
wtf I have no idea how this built and worked on Linux. I'll take a look when I'm on my computer next. Unblocks https://github.com/fgsfdsfgs/perfect_dark/pull/389#issuecomment-2384121638